tf2onnx/shape_inference.py

```diff
@@ -115,43 +116,18 @@ def infer_shape_for_node(g, node):
             return False
         return set_shape_from_input(g, shape_node.input[0], node.output[0])
 
-    if node.type == "ConcatV2":
-        axis_node = node.inputs[-1]
-        if not axis_node.is_const():
-            return False
-
-        axis = axis_node.get_tensor_value()
-        val = 0
-        data_inputs = node.input[:-1]
-        for i in data_inputs:
-            s = g.get_shape(i)
-            if s is None:
-                return False
-
-            if s[axis] == -1:
-                val = -1
-                break
-            val += s[axis]
-
-        s1 = g.get_shape(node.input[0])
-        if axis < 0:
-            axis += len(s1)
-        new_shape = s1[:axis] + [val]
-        if axis < len(s1) - 1:
-            new_shape += s1[axis + 1:]
-
-        g.set_shape(node.output[0], new_shape)
-        logger.debug("set ConcatV2 node [%s] with new shape %s", node.output[0], new_shape)
-        return True
-
     if node.type == "Gather":
         # uses the follwing link to know how to infer shape of output
         # https://www.tensorflow.org/api_docs/python/tf/gather
         shape_params = g.get_shape(node.input[0])
         shape_indices = g.get_shape(node.input[1])
-        axis = node.input[2].get_tensor_value()
+        # in lower tf version, gather only has 2 inputs

```

Comment:
which version? please add to comment.

Reply:
https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/gather.html

Reply:
will add related link to the comment